src/_P092_DLbus.ino

```diff
@@ -0,0 +1,1180 @@
+        value = sensorvalue * 4;
+        break;
+      case ROOM:
+        ReadData.mode = (sensorvalue & 0x600) >> 9;
+        value = (sensorvalue & 0x1ff) * 0.1;
+        break;
+      default:
+        return false;
+    }
+  }
+  else { // sign negative
+    sensorvalue |= 0xf000;
+    // calculations for different sensor types
+    switch (sensortype) {
+      case DIGITAL:
+        value = true;
+        break;
+      case TEMP:
+        value = (sensorvalue - 65536) * 0.1;
+        break;
+      case RAYS:
+        value = sensorvalue - 65536;
+        break;
+      case VOLUME_FLOW:
+        value = (sensorvalue - 65536) * 4;
+        break;
+      case ROOM:
+        ReadData.mode = (sensorvalue & 0x600) >> 9;
+        value = ((sensorvalue & 0x1ff) - 65536) * 0.1;
+        break;
+      default:
+        return false;
+    }
+  }
+  ReadData.value = value;
+  return true;
+}
+
+boolean P092_fetch_output(int number) {
+  int outputs;
+
+  if (DataSettings.OutputBytes > 1)
+    outputs = (ByteStream[DataSettings.IdxOutput + 1] << 8) | ByteStream[DataSettings.IdxOutput];
+  else
+    outputs = ByteStream[DataSettings.IdxOutput];
+  ReadData.value = (!!(outputs & (1 << (number - 1))));
+  return true;
+}
+
+boolean P092_fetch_speed(int number) {
+  byte speedbyte;
+
+  speedbyte = ByteStream[DataSettings.IdxDrehzahl + (number - 1)];
+  if ((speedbyte & 0x80) == 0x80)
+    return false;
+  ReadData.value = (speedbyte & 0x1f);
+  return true;
+}
+
+boolean P092_fetch_analog(int number) {
+  byte analogbyte;
+
+  analogbyte = ByteStream[DataSettings.IdxAnalog + (number - 1)];
+  if ((analogbyte & 0x80) == 0x80)
+    return false;
+  ReadData.value = (analogbyte * 0.1);
+  return true;
+}
+
+boolean P092_CheckHmRegister(int number) {
+  switch (number) {
+    case 1:
+      if ((ByteStream[DataSettings.IdxHmRegister] & 0x1) != 0x01)
+        return false;
+      power_index = DataSettings.IdxHeatMeter1;
+      kwh_index = DataSettings.IdxkWh1;
+      mwh_index = DataSettings.IdxMWh1;
+      break;
+    case 2:
+      if ((ByteStream[DataSettings.IdxHmRegister] & 0x2) != 0x02)
+        return false;
+      power_index = DataSettings.IdxHeatMeter2;
+      kwh_index = DataSettings.IdxkWh2;
+      mwh_index = DataSettings.IdxMWh2;
+      break;
+    case 3:
+      if ((ByteStream[DataSettings.IdxHmRegister] & 0x4) != 0x04)
+        return false;
+      power_index = DataSettings.IdxHeatMeter3;
+      kwh_index = DataSettings.IdxkWh3;
+      mwh_index = DataSettings.IdxMWh3;
+      break;
+    default:
+      return false;
+  }
+  return true;
+}
+
+boolean P092_fetch_heatpower(int number) {
+  // current power
+  int high;
+
+  if (P092_CheckHmRegister(number) == false)
+    return false;
+  byte b1 = ByteStream[power_index];
+  byte b2 = ByteStream[power_index + 1];
+  if (DataSettings.CurrentHmBytes > 2) {
+    byte b3 = ByteStream[power_index + 2];
+    byte b4 = ByteStream[power_index + 3];
+    high = 65536 * b4 + 256 * b3 + b2;
```

Comment:
65536 is too much for `int high`

Reply:
Thanks, changed to int32_t

Reply:
int was already 32 bit, but making it `int32_t` is much more clear what it is as is shown here :)